SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/fertilizer.py

```diff
@@ -0,0 +1,258 @@
+from typing import Optional, Dict
+from math import log, exp
+
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+from SC_redesign.Crop_and_Soil.crop_and_soil_constants import HECTARES_TO_SQUARE_MILLIMETERS, \
+    LITERS_TO_CUBIC_MILLIMETERS, KILOGRAMS_TO_MILLIGRAMS
+
+"""
+This module incorporates equations from the SurPhos model to simulate the leaching of Phosphorus from fertilizer applied
+to the soil surface being absorbed into the soil and/or removed from the field by runoff.
+"""
+
+
+class Fertilizer:
+
+    def __init__(self, soil_data: Optional[SoilData] = None):
+        self.data = soil_data or SoilData()  # initialize with defaults, if not given
+
+    def do_fertilizer_phosphorus_operations(self, rainfall: float, runoff: float, field_size: float) -> None:
+        """The main routine that updates phosphorus in surface-applied fertilizer on a daily basis.
+
+        Args:
+            rainfall: amount of rainfall on this day (mm)
+            runoff: amount of runoff on this day (mm)
+            field_size: size of the field (ha)
+
+        """
```

Comment:
Should we start using Numpy style here? I think this is the way:

```suggestion
        """The main routine that updates phosphorus in surface-applied fertilizer on a daily basis.

        Parameters
        -------------
        rainfall : float
            amount of rainfall on this day (mm) 
        runoff : float 
            amount of runoff on this day (mm)
        field_size : float
            size of the field (ha)
        """
```

I'll just go ahead and say it: I don't like this. It is not efficient to have to write the type of the argument when that information already exists in the signature. Other sphinx types do this automatically but it seems Numpy style needs it explicitly (although type hints are not included in Numpy method signatures according to [this example doc](https://numpydoc.readthedocs.io/en/latest/example.html#example)).

Reply:
Yeah I'm pretty sure NumPy style also likes optional or defaults specified after the type, which is yet another thing to repeat from the signature. All new modules that I start will use NumPy style; this one was already started with Google style and I didn't want to make it a hodgepodge of different styles.

Reply:
Agreed